pkg/v5/resource/chart/create.go

```diff
@@ -79,7 +79,7 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		//
 		err = guestHelmClient.InstallFromTarball(tarballPath, chartOperatorNamespace,
 			helm.ReleaseName(createState.ReleaseName),
-			helm.ValueOverrides([]byte("{}")),
+			helm.ValueOverrides([]byte("image.registry="+r.registryDomain)),
```

Comment:
Just curious here. Before we were passing JSON but now the input format is different. Is this an alternative format?

Reply:
+1 Ross. I don't follow here, this does not much my knowledge about the helm client.

Reply:
The format in this PR is the proper format AFAICT, `ValueOverrides` is called from the command line `install`, `update`, etc. commands, for taking the values from `--set` options, see https://github.com/kubernetes/helm/blob/master/docs/charts.md#using-the-cli-with-tags-and-conditions. Those are later coalesced here https://github.com/kubernetes/helm/blob/master/pkg/chartutil/requirements.go#L294 and put to YAML here https://github.com/kubernetes/helm/blob/master/pkg/chartutil/requirements.go#L299. The same process is taken when we call `InstallRelease` on the client.

As mentioned in parts of Helm docs, YAML is a superset of JSON, but not sure if setting here generic JSON strings will work, have you tested it?

Reply:
Ah OK got it. Makes sense that it's YAML. It was the YAML and the `+r.registryDomain` that confused me but that is just string concatenation.

For the config map changes we pass in a nested struct which is converted to JSON. That is working fine but calling `helm get values` returns the same data as YAML.

https://github.com/giantswarm/cluster-operator/blob/4eddf4c98a28d8024d24337c9b76d4f9f4cef362/pkg/v5/configmap/desired.go#L29-L35

I'd prefer it if we settled on one input format for passing values to Helm. YAML does have a superset of features over JSON. But I'm not sure we need those and I think JSON is easier to work with in the operators. Not a blocker though so go ahead when ready.

Reply:
I agree that we should be consistent here. This is also relevant in the `release-operator` story because currently I have to convert YAML from the `installations` repo to JSON for to align with the `chart-operator` config changes. 

I am fine with either, I have no preference of YAML over JSON - consistency would be nice though.

Reply:
Agree, let's go for JSON then, thanks!

Reply:
Done, PTAL.

Reply:
Thanks. I like the JSON handling.